File: xion-android-demo/app/src/main/java/com/burnt/xiondemo/data/model/TransactionResult.kt
package com.burnt.xiondemo.data.model

import kotlinx.serialization.Serializable

@Serializable
data class TransactionResult(
    val txHash: String,
    val success: Boolean,
    val gasUsed: String,
    val gasWanted: String,
    val height: Long,
    val rawLog: String,
    val timestamp: String = "",
    val fee: String = "",
    val txType: String = "",
    val amount: String = "",
    val amountDenom: String = "",
    val recipient: String = ""
)

File: xion-ios/XionDemo/Models/TransactionResult.swift
import Foundation

struct TransactionResult: Identifiable, Codable {
    var id: String { txHash }

    let txHash: String
    let success: Bool
    let gasUsed: String
    let gasWanted: String
    let height: Int64
    let rawLog: String
    var timestamp: String = ""
    var fee: String = ""
    var txType: String = ""
    var amount: String = ""
    var amountDenom: String = ""
    var recipient: String = ""
    // Brale-sourced rows: human status (Pending/Processing/Completed/...), a
    // preformatted amount (e.g. "$1.00") shown instead of the micro-unit `amount`,
    // and whether the transfer is still settling.
    var status: String = ""
    var displayAmount: String = ""
    var inProgress: Bool = false
}
